pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -137,6 +132,103 @@ def relative_path_for_content_artifact(self):
         return (Package(**pkg_dict), None)
 
 
+class Pulp2Srpm(Pulp2to3Content):
+    """
+    Pulp 2to3 detail content model to store Pulp 2 SRPM content details for Pulp 3 content creation.
+    """
+
+    name = models.TextField()
+    epoch = models.TextField()
+    version = models.TextField()
+    release = models.TextField()
+    arch = models.TextField()
+    checksum = models.TextField()
+    checksumtype = models.TextField()
+
+    repodata = JSONField(dict)
+    is_modular = models.BooleanField(default=False)
+    size = models.PositiveIntegerField()
+    filename = models.TextField()
+
+    pulp2_type = 'srpm'
+
+    class Meta:
+        unique_together = (
+            'name', 'epoch', 'version', 'release', 'arch', 'checksumtype', 'checksum',
+            'pulp2content')
+        default_related_name = 'srpm_detail_model'
+
+    @property
+    def expected_digests(self):
+        """Return expected digests."""
+        return {self.checksumtype: self.checksum}
+
+    @property
+    def expected_size(self):
+        """Return expected size."""
+        return self.size
+
+    @property
+    def relative_path_for_content_artifact(self):
+        """Return relative path."""
+        return self.filename
+
+    @classmethod
+    async def pre_migrate_content_detail(cls, content_batch):
+        """
+        Pre-migrate Pulp 2 SRPM content with all the fields needed to create a Pulp 3 Package.
+
+        Args:
+             content_batch(list of Pulp2Content): pre-migrated generic data for Pulp 2 content.
+
+        """
+
+        pulp2_id_obj_map = {pulp2content.pulp2_id: pulp2content for pulp2content in content_batch}
+        pulp2_ids = pulp2_id_obj_map.keys()
+        pulp2_content_batch = pulp2_models.SRPM.objects.filter(id__in=pulp2_ids).as_pymongo().only(
+            'name',
+            'epoch',
+            'version',
+            'release',
+            'arch',
+            'checksum',
+            'checksumtype',
+            'repodata',
+            'size',
+            'filename',
+            'pk',
+        )
+
+        pulp2srpm_to_save = []
+        for srpm in pulp2_content_batch:
+            decompressed_repodata = _decompress_repodata(srpm['repodata'])
+            srpm['repodata'] = decompressed_repodata
+
+            pulp2srpm_to_save.append(
+                cls(name=srpm['name'],
+                    epoch=srpm['epoch'],
+                    version=srpm['version'],
+                    release=srpm['release'],
+                    arch=srpm['arch'],
+                    checksum=srpm['checksum'],
+                    checksumtype=srpm['checksumtype'],
+                    repodata=srpm['repodata'],
+                    size=srpm['size'],
+                    filename=srpm['filename'],
+                    pulp2content=pulp2_id_obj_map[srpm['_id']])
+            )
+        cls.objects.bulk_create(pulp2srpm_to_save, ignore_conflicts=True)
```

Comment:
@dralley @goosemania i could not subclass from Pulp2Rpm because bulk_create does not work with multi-table inheritance, unless we make Pulp2Srpm a proxy model, which i think we do not want to. https://github.com/django/django/blob/master/django/db/models/query.py#L457-L478 or we call save() on each item

Reply:
In theory, I think we can make a proxy model Pulp2RpmBase and subclass it for both, Pulp2Srpm and Pulp2Rpm.
But I'm not sure it's wroth a hassle. 
@dralley , what do you think?

Reply:
Do we need the concrete base type?  If not, an abstract class would work fine and avoid proxy/multi-table inheritance behavior